src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -127,91 +127,99 @@ def send_alert_webhook(
+        )
+
+        if not nodedata:
+            extra = {
+                "event_id": instance_id,
+                "occurrence_id": occurrence_id,
+                "rule": rule,
+                "sentry_app": sentry_app.slug,
+                "group_id": group_id,
+            }
+            raise SentryAppSentryError(
+                message=SentryAppWebhookFailureReason.MISSING_EVENT, webhook_context=extra
+            )
 
-    data = {"event": event_context, "triggered_rule": rule}
+        occurrence = None
+        if occurrence_id:
+            occurrence = IssueOccurrence.fetch(occurrence_id, project_id=project.id)
 
-    # Attach extra payload to the webhook
-    if additional_payload_key and additional_payload:
-        data[additional_payload_key] = additional_payload
+            if not occurrence:
+                raise SentryAppSentryError(
+                    message=SentryAppWebhookFailureReason.MISSING_ISSUE_OCCURRENCE,
+                    webhook_context={"occurrence_id": occurrence_id, "project_id": project.id},
+                )
 
-    request_data = AppPlatformEvent(
-        resource="event_alert", action="triggered", install=install, data=data
-    )
+        group_event = GroupEvent(
+            project_id=project.id,
+            event_id=instance_id,
+            group=group,
+            data=nodedata,
+            occurrence=occurrence,
+        )
+
+        event_context = _webhook_event_data(group_event, group.id, project.id)
 
-    send_and_save_webhook_request(sentry_app, request_data)
+        data = {"event": event_context, "triggered_rule": rule}
 
-    # On success, record analytic event for Alert Rule UI Component
-    if request_data.data.get("issue_alert"):
-        analytics.record(
-            "alert_rule_ui_component_webhook.sent",
-            organization_id=organization.id,
-            sentry_app_id=sentry_app_id,
-            event=f"{request_data.resource}.{request_data.action}",
+        # Attach extra payload to the webhook
+        if additional_payload_key and additional_payload:
+            data[additional_payload_key] = additional_payload
+
+        request_data = AppPlatformEvent(
+            resource="event_alert", action="triggered", install=install, data=data
         )
 
+        try:
+            send_and_save_webhook_request(sentry_app, request_data)
+        except (ApiHostError, ApiTimeoutError, RequestException, ClientError):
+            # record success as the preperation portion did not fail
+            lifecycle.record_success()
+            raise
+
+        # On success, record analytic event for Alert Rule UI Component
+        if request_data.data.get("issue_alert"):
+            analytics.record(
+                "alert_rule_ui_component_webhook.sent",
+                organization_id=organization.id,
+                sentry_app_id=sentry_app_id,
+                event=f"{request_data.resource}.{request_data.action}",
+            )
+
```

Comment:
i would move this block outside of the context manager block (e.g. unindent) if it's not related to preparation, then you don't need to record success if `send_and_save_webhook_request` fails
i think you'd also be able to remove catching `ApiHostError` etc since you don't have to record success for them, and let the `retry_decorator` do the work